nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -248,58 +219,59 @@ def main() -> None:
             # Let systemd-boot attempt an installation if a previous one wasn't found
             needs_install = True
         else:
-            sdboot_version = f'({m.group(2)})'
+            sdboot_version = f"({m.group(2)})"
             if systemd_version != sdboot_version:
-                print("updating systemd-boot from %s to %s" % (sdboot_version, systemd_version))
+                print(f"updating systemd-boot from {sdboot_version} to {systemd_version}")
                 needs_install = True
 
         if needs_install:
             subprocess.check_call(["@systemd@/bin/bootctl", "--path=@efiSysMountPoint@", "update"])
 
-    mkdir_p("@efiSysMountPoint@/efi/nixos")
-    mkdir_p("@efiSysMountPoint@/loader/entries")
+    os.makedirs("@efiSysMountPoint@/efi/nixos", exist_ok=True)
+    os.makedirs("@efiSysMountPoint@/loader/entries", exist_ok=True)
 
     gens = get_generations()
     for profile in get_profiles():
         gens += get_generations(profile)
     remove_old_entries(gens)
     for gen in gens:
         try:
-            write_entry(*gen, machine_id)
-            for specialisation in get_specialisations(*gen):
-                write_entry(*specialisation, machine_id)
-            if os.readlink(system_dir(*gen)) == args.default_config:
-                write_loader_conf(*gen)
+            write_entry(gen, machine_id)
+            for specialisation in get_specialisations(gen):
+                write_entry(specialisation, machine_id)
+            if os.readlink(system_dir(gen)) == args.default_config:
+                write_loader_conf(gen)
         except OSError as e:
-            print("ignoring profile '{}' in the list of boot entries because of the following error:\n{}".format(profile, e), file=sys.stderr)
+            profile = f"profile '{gen.profile}'" if gen.profile else "default profile"
```

Comment:
In certain scenarios (I may have done a boo boo), the profile may always be none:

```
ignoring generation 'None' in the list of boot entries because of the following error:
1
ignoring generation 'None' in the list of boot entries because of the following error:
2
ignoring generation 'None' in the list of boot entries because of the following error:
3
ignoring generation 'None' in the list of boot entries because of the following error:
4
ignoring generation 'None' in the list of boot entries because of the following error:
5
ignoring generation 'None' in the list of boot entries because of the following error:
6
```